File: backend/src/email/email.controller.ts
import { Controller, Post, Body } from '@nestjs/common';
import { EmailService } from './email.service';

@Controller('email')
export class EmailController {
  constructor(private readonly emailService: EmailService) {}

  @Post('send')
  async sendEmail(
    @Body() body: { to: string; subject: string; html: string },
  ) {
    const result = await this.emailService.sendEmail(body);
    return { success: result };
  }

  @Post('distribute')
  async distributeExam(
    @Body()
    body: {
      examTitle: string;
      pdfHtml: string;
      colleges: { id: string; name: string; email: string }[];
    },
  ) {
    return this.emailService.distributeExamToColleges(
      body.examTitle,
      body.pdfHtml,
      body.colleges,
    );
  }

  @Post('notify-modification')
  async notifyModification(
    @Body()
    body: {
      examTitle: string;
      modifications: { changeType: string; description: string }[];
      colleges: { id: string; name: string; email: string }[];
    },
  ) {
    return this.emailService.sendModificationNotification(
      body.examTitle,
      body.modifications,
      body.colleges,
    );
  }
}

File: backend-spring/src/main/java/com/bloomgate/email/EmailController.java
package com.bloomgate.email;

import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/email")
public class EmailController {

    private final EmailService emailService;

    public EmailController(EmailService emailService) {
        this.emailService = emailService;
    }

    @PostMapping("/send")
    public Map<String, Object> sendEmail(@RequestBody Map<String, Object> body) {
        String to = (String) body.get("to");
        String subject = (String) body.get("subject");
        String html = (String) body.get("html");
        boolean result = emailService.sendEmail(to, subject, html);
        return Map.of("success", result);
    }

    @PostMapping("/distribute")
    public List<Map<String, Object>> distributeExam(@RequestBody Map<String, Object> body) {
        String examTitle = (String) body.get("examTitle");
        String pdfHtml = (String) body.get("pdfHtml");
        @SuppressWarnings("unchecked")
        List<Map<String, Object>> colleges = (List<Map<String, Object>>) body.get("colleges");
        return emailService.distributeExamToColleges(examTitle, pdfHtml, colleges);
    }

    @PostMapping("/notify-modification")
    public List<Map<String, Object>> notifyModification(@RequestBody Map<String, Object> body) {
        String examTitle = (String) body.get("examTitle");
        @SuppressWarnings("unchecked")
        List<Map<String, Object>> modifications = (List<Map<String, Object>>) body.get("modifications");
        @SuppressWarnings("unchecked")
        List<Map<String, Object>> colleges = (List<Map<String, Object>>) body.get("colleges");
        return emailService.sendModificationNotification(examTitle, modifications, colleges);
    }
}
